spec/src/main/asciidoc/02-config.adoc

```diff
@@ -3,3 +3,24 @@
 == Introduction
 
 
+== Support Injection
+A ConfigMapping interface can be injected via the `@Inject`. The following example demonstrates the usage.
+
+[source]
+----
+customer.name=Bob
+customer.age=28
+----
+Use the following code snippet to retrieve the above configuration.
+[source]
+----
+@ConfigMapping("customer")
+public interfact MyConfigMapping{
+    String name;
+    int age;
+}
+
+@Inject MyConfigMapping myConfigMapping;
+----
```

Comment:
Usage examples are not normative. We should definitely have usage examples but they should be clearly marked as non-normative examples somehow. Maybe we can use admonition blocks:

```asciidoc
[Usage example]
===
The following is a usage example for how `@Inject` may be used:

[source,java]
----
@ConfigMapping("customer")
public interfact MyConfigMapping{
    String name;
    int age;
}
@Inject MyConfigMapping myConfigMapping;
----
===

```